### ISSUE TITLE
CSV newspapers toolchain

### ISSUE THREAD

**Issue by mjordan** [MarcusBarnes/mik#216]
Time to move on to additional CSV toolchains. One for newspapers is as good a place to start as any, since the CONTENTdm newspapers toolchain has been so thoroughly tested and the CSV toolchain can probably be just a simplification of that one.

**Comment by mjordan** [comment]
Wiki page started at https://github.com/MarcusBarnes/mik/wiki/Toolchain:-CSV-newspapers.

Only real outstanding issue is that specifying MODS or OBJ in the `datastreams[]` option doesn't work yet.

**Comment by mjordan** [comment]
@mbarnes if you have time to smoke test this new toolchain I'd appreciate it. Note that the resulting ingest packages only contain the issue-level MODS.xml file and a TIFF and a MODS.xml file for each page. We can look at adding derivatives later using post-write hooks.

**Comment by mjordan** [comment]
Sample .ini file

```
; MIK configuration file for generating Islandora newspaper ingest packages
; from a CSV metadata file and locally stored TIFFs.

[CONFIG]
config_id = CSVNewspapersTest
last_updated_on = "2016-05-14"
last_update_by = "mj"

[FETCHER]
class = Csv
input_file = '/home/mark/Downloads/non_cdm_newspaper_samples.csv'
temp_directory = "/tmp/csv_newspapers_temp"
; The column heading in the CSV file that contains the name of the unique ID for each row.
record_key = Identifier

[METADATA_PARSER]
class = mods\CsvToMods
mapping_csv_path = 'csv_newspapers_test_mappings.csv'
temp_directory = "/tmp/csv_newspapers_temp"

[FILE_GETTER]
class = CsvNewspapers
temp_directory = "/tmp/csv_newspapers_temp"
input_directory = "/home/mark/Downloads/non_cdm_newspaper_samples"
; The column heading in the CSV file that contains the name of the directory where the page files for the issue are.
file_name_field = Directory

[WRITER]
class = CsvNewspapers
output_directory = "/tmp/csv_newspapers_output"
; Must be MODS.xml
metadata_filename = MODS.xml
; datastreams[] = MODS

[MANIPULATORS]
metadatamanipulators[] = "FilterModsTopic|subject"

[LOGGING]
path_to_log = "/tmp/csv_newspapers_output/mik.log"
path_to_manipulator_log = "/tmp/csv_newspapers_output/mik_manipulator.log"
```

**Comment by MarcusBarnes** [comment]
@mjordan Initial tests worked with both flat and hierarchical directories containing issue master page images.

**Comment by mjordan** [comment]
Great, thanks. If you have no objections, feel free to merge that branch into master and I'll link the wiki page from the table of contents. I can update the README as well.

**Comment by MarcusBarnes** [comment]
@mjordan No objections.  I'll create a pull-request and merge momentarily.

**Comment by MarcusBarnes** [comment]
Closed with commit fbd3a82.  Thank you @mjordan for your contribution.
